Create CLI documentation with a Rust script (instead of Python) + style improvement

heeckhau opened the pull request:
I noticed that the CLI documentation was being generated using a Python script, so I rewrote it in Rust to make it easier to understand and more maintainable. Although it’s not a 1-to-1 conversion, I aimed to stay close to the original implementation to facilitate the review process. The output remains identical.

Additionally, I made a small change to the Markdown template (in a separate commit) to enhance the appearance of the generated documentation:
![image](https://github.com/user-attachments/assets/262e41b0-b46c-4641-8655-837c257efce8)

onbjerg reviewed (approved):
personally love this, we should give some love to the cargo script feature. curious what others think cc @mattsse @DaniPopes @rkrasiuk @Rjected

DaniPopes reviewed (approved):
Yeah, this is way better, thanks!

This was adapted from the Foundry book (<https://github.com/foundry-rs/book/blob/8f8215a10248f8e4d4870e3c03dd836d64169fd1/scripts/gen_output/help.py#L1>), would you mind also opening a PR there too?

DaniPopes commented on book/cli/help.rs:
```diff
@@ -0,0 +1,311 @@
+#!/usr/bin/env cargo +nightly -Zscript
```
Needs `-S` passed to `env` for multiple arguments

heeckhau replied:
Added

Rjected reviewed (commented):
epic, looks like the current book CI lint uses a stable toolchain, so you will have to install the nightly toolchain as well in the lint workflow

heeckhau commented:
This change was merged in Foundry: https://github.com/foundry-rs/book/pull/1281
@DaniPopes I have updated the PR to include your changes.

I've also added Rust nightly to the GitHub workflow file
